infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenshiftRuntimeContext.java

```diff
@@ -0,0 +1,77 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.workspace.infrastructure.openshift;
+
+import com.google.inject.assistedinject.Assisted;
+
+import org.eclipse.che.api.core.ValidationException;
+import org.eclipse.che.api.core.model.workspace.config.Environment;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.installer.server.InstallerRegistry;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.api.workspace.server.spi.InternalInfrastructureException;
+import org.eclipse.che.api.workspace.server.spi.InternalRuntime;
+import org.eclipse.che.api.workspace.server.spi.RuntimeContext;
+import org.eclipse.che.api.workspace.server.spi.RuntimeInfrastructure;
+import org.eclipse.che.workspace.infrastructure.openshift.environment.OpenshiftEnvironment;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriBuilderException;
+import java.net.URI;
+
+import static org.eclipse.che.api.workspace.server.OutputEndpoint.OUTPUT_WEBSOCKET_ENDPOINT_BASE;
+
+
+/**
+ * @author Sergii Leshchenko
+ */
+public class OpenshiftRuntimeContext extends RuntimeContext {
+    private final OpenshiftEnvironment  openshiftEnvironment;
+    private final RuntimeContextFactory runtimeContextFactory;
+    private final String                apiEndpoint;
+
+    @Inject
+    public OpenshiftRuntimeContext(@Assisted Environment environment,
+                                   @Assisted OpenshiftEnvironment openshiftEnvironment,
+                                   @Assisted RuntimeIdentity identity,
+                                   @Assisted RuntimeInfrastructure infrastructure,
+                                   InstallerRegistry installerRegistry,
+                                   RuntimeContextFactory runtimeContextFactory,
+                                   @Named("che.api") String apiEndpoint) throws ValidationException,
+                                                                                InfrastructureException {
+        super(environment, identity, infrastructure, installerRegistry);
+        this.runtimeContextFactory = runtimeContextFactory;
+        this.openshiftEnvironment = openshiftEnvironment;
+        this.apiEndpoint = apiEndpoint;
+    }
+
+    @Override
+    public URI getOutputChannel() throws InfrastructureException {
+        try {
+            final URI apiURI = URI.create(apiEndpoint);
+            return UriBuilder.fromUri(apiURI)
+                             .scheme("https".equals(apiURI.getScheme()) ? "wss" : "ws")
+                             .replacePath(apiURI.getPath().replace("/api", ""))
+                             .path(OUTPUT_WEBSOCKET_ENDPOINT_BASE)
```

Comment:
Introduced new property `che.websocket.endpoint.base` and added using it while construction of output channel